development/openstack-base-focal-wallaby/README.md

```diff
@@ -1,269 +1,371 @@
+* 8 GiB memory
+* Enough CPU cores to support your workload
+* Two disks
+* Two cabled network interfaces
 
-For example, a section similar to this exists in the bundle.yaml file.  The third "column" are the values to set.  Some servers may not have eno2, they may have something like eth2 or some other network device name.  This needs to be adjusted prior to deployment.  The same principle holds for osd-devices.  The third column is a whitelist of devices to use for Ceph OSDs.  Adjust accordingly by editing bundle.yaml before deployment.
+The first disk is used for the node's operating system, and the second is for
+Ceph storage.
 
-```
-variables:
-  openstack-origin:    &openstack-origin     distro
-  data-port:           &data-port            br-ex:eno2
-  worker-multiplier:   &worker-multiplier    0.25
-  osd-devices:         &osd-devices          /dev/sdb /dev/vdb
-```
+The first network interface is used for communication between cloud services
+(East/West traffic), and the second is for network traffic between the cloud
+and all external networks (North/South traffic).
 
-Servers should have:
+> **Important**: The three MAAS nodes are needed for the actual OpenStack
+  cloud; they do not include the Juju controller. You actually need a minimum
+  of four nodes. The controller node however can be a smaller system (1 CPU and
+  4 GiB memory). Juju [constraints][juju-constraints-controller] can be used to
+  target this smaller system at controller-creation time.
 
- - A minimum of 8GB of physical RAM.
- - Enough CPU cores to support your capacity requirements.
- - Two disks (identified by /dev/sda and /dev/sdb); the first is used by MAAS for the OS install, the second for Ceph storage.
- - Two cabled network ports on eno1 and eno2 (see below).
+## Topology
 
-Servers should have two physical network ports cabled; the first is used for general communication between services in the Cloud, the second is used for 'public' network traffic to and from instances (North/South traffic) running within the Cloud.
+* 3 MAAS nodes, with each hosting one of the following:
+    * Ceph storage
+    * Nova Compute
+    * NTP
 
-## Components
+* LXD containers for the following (distributed among the 3 MAAS nodes):
+    * Ceph monitors (x3)
+    * Ceph RADOS Gateway
+    * Cinder
+    * Glance
+    * Horizon
+    * Keystone
+    * MySQL8 (x3)
+    * Neutron
+    * Nova Cloud Controller
+    * OVN (x3)
+    * Placement
+    * RabbitMQ
+    * Vault
 
- - 3 Nodes for Nova Compute and Ceph, with RabbitMQ, MySQL, Keystone, Glance, Neutron, OVN, Nova Cloud Controller, Ceph RADOS Gateway, Cinder and Horizon under LXC containers.
+## Download the bundle
 
-All physical servers (not LXC containers) will also have NTP installed and configured to keep time in sync.
+If not already done, clone the [openstack-bundles][openstack-bundles]
+repository:
 
-## Deployment
+    git clone https://github.com/openstack-charmers/openstack-bundles ~/openstack-bundles
```

Comment:
This seems self-referential. If you are reading this README you have probably done this. And if you are reading it on GitHub we should not need to tell you how to clone a repository.

Reply:
People can stumble upon it on GitHub and may not have the knowledge. It may also be viewed in the Charm Store.